src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java

```diff
@@ -176,6 +186,18 @@ public CalledHaplotypes callMutations(
         return new CalledHaplotypes(outputCallsWithEventCountAnnotation, calledHaplotypes);
     }
 
+    // check whether two alleles coming from different variant contexts and with possibly different reference alleles
+    // could in fact be the same.  The condition is that one is a prefix of the other
+    private boolean allelesAreConsistent(final Allele allele1, final Allele allele2) {
```

Comment:
Gonna have to think about this more but don't want to delay handing the review back to you. It seems that the way it's written,

VC: A -> AA
Given allele: A -> AAA


are deemed consistent and therefore emitted forcibly...I don't think that's what we want

Reply:
You are right that this happens and I was actually debating myself about the same thing.  I was willing to be generous and possibly admit an extra allele at a GGA site because I didn't want to complicate the code and re-represent the GGA allele in the merged variant context.

However, I realize now that without complicating the code I can simply add the condition that the difference between the ref and alt lengths agree.